### PR TITLE
Add acra-rotate cli tools docs

### PR DESCRIPTION
**Pull request by ZhmakAS**

### CODE REVIEW

File: acra/configuring-maintaining/general-configuration/acra-rotate.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+  Generate markdown file with text description of all flags.
+  Output file is `configs/markdown_acra-addzone.md`.
+  Works in pair with `--dump_config`.
+
+### Network
+
+* `--db_connection_string=<db_connection_url>`
+
+  Connection string to DB.
```
Comment (Lagovas) [inline acra/configuring-maintaining/general-configuration/acra-rotate.md:64]: will be great to add exapmle of connection string or describe expected format

File: acra/configuring-maintaining/general-configuration/acra-rotate.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  Path to YAML configuration file.
+
+* `--dump_config`
+
+  Dump configuration to `configs/acra-addzone.yaml`.
+
+* `--generate_markdown_args_table`
+
+  Generate markdown file with text description of all flags.
+  Output file is `configs/markdown_acra-addzone.md`.
```
Comment (Lagovas) [inline acra/configuring-maintaining/general-configuration/acra-rotate.md:57]:
```suggestion
  Output file is `configs/markdown_acra-rotate.md`.
```

File: acra/configuring-maintaining/general-configuration/acra-rotate.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+### Configuration files
+
+* `--config_file=<filename>`
+
+  Path to YAML configuration file.
+
+* `--dump_config`
+
+  Dump configuration to `configs/acra-addzone.yaml`.
```
Comment (Lagovas) [inline acra/configuring-maintaining/general-configuration/acra-rotate.md:52]:
```suggestion
  Dump configuration to `configs/acra-rotate.yaml`.
```

File: acra/configuring-maintaining/general-configuration/acra-rotate.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+  Use TLS to encrypt transport with HashiCorp Vault.
+  Default is `false`.
+
+
+## Usage example
+
+Acra encrypts data using storage encryption keys and these keys support rotation process, which can be done in response to a potential leak or compromise.
+
+`acra-rotate` generates new storage keypair, reads data from database via AcraServer, decrypts it and re-encrypts with new keys. In this example, new keys overwrite previous ones, so it's important to rotate all data at the same time.
```
Comment (Lagovas) [inline acra/configuring-maintaining/general-configuration/acra-rotate.md:140]: are you sure that they overwrite previous, not moving them into archive folder and use for decryption only?

File: acra/configuring-maintaining/general-configuration/acra-rotate.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+In this example, we are not going to use [Zones]({{< ref "/acra/security-controls/zones.md" >}}) for rotation example. You'll get output with base64 of new public storage key:
+
+```
+$ acra-rotate 
+--db_connection_string="postgresql://test:test@acra-client/test" \
+--zonemode_enable=false \
+--postgresql_enable \
+--sql_select="select id, 'user0'::bytea, email from users" \
+--sql_update='update users set email=$1 where id=$2'
```
Comment (Lagovas) [inline acra/configuring-maintaining/general-configuration/acra-rotate.md:159]: hm, according to parameter description we should use `?` as placeholder...

File: acra/configuring-maintaining/general-configuration/acra-rotate.md (hunk 4 above)
Comment (ZhmakAS) [reply acra/configuring-maintaining/general-configuration/acra-rotate.md:140]: Technically speaking, yes it overwrites, but you are right it worth saying that new keys will be moved to `.old` folder for decryption purposes

File: acra/configuring-maintaining/general-configuration/acra-rotate.md [hunk 6]
```diff
@@ -30,12 +30,14 @@ weight: 10
...
-  Other columns will be passed into insert/update query into placeholders.
+  Select query with placeholders where last columns in result must be ClientId/ZoneId and AcraStruct. 
+  Other columns will be passed into Insert/Update query into placeholders.
+  PostgreSQL placeholder is `$1` and MySQL is `?`.
 
 * `--sql_update=<query>`
 
-  Insert/Update query with `?` as placeholder where into first will be placed rotated AcraStruct.
+  Insert/Update query with placeholders where rotated AcraStruct will be placed into the first.
+  PostgreSQL placeholder is `$1` and MySQL is `?`.
```
Comment (Lagovas) [inline acra/configuring-maintaining/general-configuration/acra-rotate.md:40]: actually, not `$1` but `$n`. And maybe will be better to add references to source of such requirements: db drivers.
https://github.com/lib/pq - driver responsible for postgresql query execution and https://github.com/go-sql-driver/mysql for MySQL queries.

**Review by Lagovas — APPROVED** [review]
lgtm

**Pull request merged**